client/state/current-user/email-verification/actions.js

```diff
@@ -0,0 +1,27 @@
+/**
+ * Internal dependencies
+ */
+import {
+	EMAIL_VERIFY_REQUEST,
+	EMAIL_VERIFY_REQUEST_SUCCESS,
+	EMAIL_VERIFY_REQUEST_FAILURE,
+	EMAIL_VERIFY_STATE_RESET,
+} from 'state/action-types';
+import wpcom from 'lib/wp';
+
+export function verifyEmail() {
```

Comment:
Would you be interested in using the new [data layer](https://github.com/Automattic/wp-calypso/tree/master/client/state/data-layer)? We're trying to avoid using action thunks for new pieces of code. This should simplify state, and make testing a bit easier.

This example should be pretty straightforward, but feel free to ping me or @dmsnell for help.